### PR TITLE
Rework the API documentation and place the API under test

### LINKED ISSUE #5: Support CPU and GPU benchmarking on localhost
Right now CPU and GPU benchmarking in benchit take an IP address that must point to a remote machine with SSH access. Add a feature that allows the user to enter "localhost" to run on a local CPU or GPU.

Add a test to //sales/mlagility:cli.test for local CPU benchmarking.

### LINKED ISSUE #67: Don't print the "CPU/GPU is not used for ..." message on every benchmark
Right now if you run CPU or GPU benchmark you see a message like:

```
printing.log_info(
            (
                " CPU is not used for accuracy comparisons it's only used for"
                " performance comparison. So dummy inputs are used.\n"
                " User is responsible for ensuring the remote cpu server is turned on and"
                " has miniconda3 @ /home/user/ and python>=3.8 installed."
            )
        )
```

This doesn't make a lot of sense to print every single time the context of MLAgility. Clean it up!

Proposed solutions:
- Don't bother printing the "not used for accuracy comparisons..." bit
- Surpress the "User is responsible..." bit after the first successful benchmark on a device. Or only show it if there's a benchmarking error.

### LINKED ISSUE #95: Put the API under test

### LINKED ISSUE #97: Better API documentation
I propose the following changes:
- `benchmark_script()` becomes an official top-level API
  - add it to the top-level init.py
  - document it in the benchit user guide
  - move it to the API folder
- Rename the `benchit` API to `benchmark_model`
  - More descriptive about what it does
  - Disambiguates from the `benchit` CLI, which has a very different feature set
  - Matches the naming scheme of `benchmark_script`

@danielholanda thoughts?

### LINKED ISSUE #98: Clean up remote benchmarking documentation
Right now the benchit user guide mentions a `--ip` argument, however it is not implemented in the code. The code does implement a `backend` argument for the `benchit API` (aka benchmark_model() api), which is similar. We are also not sure that we want to release remote benchmarking in the first release.

Proposed solution:
- Remove the benchit user guide `--ip` argument for now. Add it back when we have fully robust remote benchmarking.
- Mention in the user guide that remote benchmarking is enabled in the code, but the feature is not under test yet and it requires SFT.
- Add a `--backend` argument that allows selection between local and cloud. Local is the default. Related to #6 

Thoughts @danielholanda?

### LINKED ISSUE #99: Put localhost benchmarking under test
For both the API and the CLI

### LINKED ISSUE #101: Clean up the benchmarking backend code
Right now the benchmarking code creates a `mlagility_cache` directory on the target machine, for both the local and remote benchmarking. This `mlagility_cache` does not mirror the layout of the official `mlagility cache` described in the user guide. It also creates some unnecessary file replication in the local case. 

All of this code could be cleaned up by unifying these cache structures, and by allowing local benchmarking to use the local `mlagility cache`.

We could potentially clean up even further by having the remote benchmarking python environment simply `pip install mlagility`, which would save us from transferring files around (versioning could be a pain though).

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #95 
Closes #97 
Closes #98 
Closes #99 
Closes #101
Closes #5 
Closes #67

### CODE REVIEW

**Review by danielholanda — APPROVED** [review]

File: src/mlagility/cli/slurm.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def run_benchit(
     cache_dir: str,
     search_dir: Optional[str] = None,
     rebuild: Optional[str] = None,
-    compiler_flags: Optional[List[str]] = None,
-    assembler_flags: Optional[List[str]] = None,
-    num_chips: Optional[int] = None,
+    groq_compiler_flags: Optional[List[str]] = None,
+    groq_assembler_flags: Optional[List[str]] = None,
+    groq_num_chips: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (jeremyfowers) [reply src/mlagility/cli/slurm.py:69]: wont fix, as discussed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,10 +28,14 @@ jobs:
           pip install pylint
           if [ -f setup.py ]; then pip install -e .; fi
           pip install transformers
+          wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+          bash Miniconda3-latest-Linux-x86_64.sh
```
[the remark this answers is not in the document]
Comment (jeremyfowers) [reply .github/workflows/test.yml]: Changed to miniconda action

File: src/mlagility/api/devices.py [hunk 3]
```diff
@@ -11,6 +11,47 @@
...
+            return os.path.join(
+                build.output_dir(self.state.cache_dir, self.state.config.build_name),
+                f"{self.device}_benchmark",
+            )
+        elif self.backend == "remote":
+            return os.path.join("/home", self.username, "mlagility_remote_cache")
+        elif self.backend == "docker":
+            return "/app"
+        else:
+            raise ValueError(f"Got backend {self.backend}, which is not allowed.")
```
Comment (danielholanda) [inline src/mlagility/api/devices.py:36]: Show allowed backends

**Review by danielholanda — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielholanda) [reply .github/workflows/test.yml]: sounds good!

File: src/mlagility/api/devices.py (hunk 3 above)
Comment (jeremyfowers) [reply src/mlagility/api/devices.py]: done!

**Pull request merged**